lib/Adapters/WowzaAdapter.js

```diff
@@ -58,6 +59,95 @@ function deleteInactiveStreams(currentlyActiveStreams) {
+    _.each(incomingStreams, (item) => {
+        let re=/([01]_[^_]+)_(?:publish_)?(.*)/.exec(item.name);
+        if (re && re.length===3) {
+            let entryId=re[1];
+
+            if (this._entryIds && !this._entryIds.has(entryId)) {
+                return;
+            }
+            let flavor=re[2];
+            if (!uniqueList[entryId])
+                uniqueList[entryId]={ 'entryId': entryId,
+                    'sourceIp': item.sourceIp,
+                    'flavors': []};
+
+            if (item.sourceIp === "local (Transcoder)") {
+                uniqueList[entryId]['flavors'].push(flavor);
+            }
+        } else {
+            logger.warn("Unidentified stream %j",item);
+        }
+    });
+}
+
+function checkFlavorsMatch(entry, flavorParamsIds) {
+    //no need for API call if nothing has changed
+    if (entry) {
+        if (!entry.canPlay) {
+            if (entry.retryCount > this.flavorMismatchRetryAttempts) {
+                entry.canPlay = true;
+                logger.info("[%s] Flavors mismatch occurred for %d retries, override configured [%j] with [%j]", entry.entryId, entry.retryCount, entry.flavorParamsIds, flavorParamsIds);
+                ///TODO should we override or just go with the configured ones?!?!?
+                entry.flavorParamsIds = flavorParamsIds;
+                delete entry.retryCount;
+            } else if (entry.flavorParamsIds === flavorParamsIds) {
+                entry.canPlay = true;
+                logger.info("[%s] Configured flavors are the same: [%j] after %d tries", entry.entryId, entry.flavorParamsIds, entry.retryCount);
+                delete entry.retryCount;
+            }
+            else {
+                logger.info("[%s] Flavors mismatch between configured [%j] and actual [%j]", entry.entryId, entry.flavorParamsIds, flavorParamsIds);
+                entry.retryCount++;
+            }
+        }
+        return true;
+    }
+    return false;
+}
+
+function retrieveEntryInfo(entryObj, entryId, flavorParamsIds, sourceIp) {
+    this.entries[entryId] = entryObj;
+    //If playing flavors are equal to the configured flavors, allow entry to start playing immediately.
+    //otherwise wait another iteration to allow flavors to arrive.
+    entryObj.canPlay = (entryObj.flavorParamsIds === flavorParamsIds);
+    if (!entryObj.canPlay) {
+        entryObj.retryCount = 1;
+        logger.info("[%s] Flavors mismatch between configured [%j] and actual [%j]", entryId, entryObj.flavorParamsIds, flavorParamsIds);
+    }
+    entryObj.sourceIp = sourceIp;
+    entryObj.baseUrl = this._baseUrl;
+    entryObj.getStreamInfo = getStreamInfo;
+    logger.info("[%s] Received stream info for entry: %j", entryId, entryObj);
+}
+
+function mapNewStream(stream) {
+    let that = this;
+    var entryId = stream.entryId;
+    let flavorParamsIds = stream.flavors.join(',');
+
+    if (checkFlavorsMatch.call(that, that.entries[entryId], flavorParamsIds)) {
```

Comment:
if (that.entries[entryId]) {
     checkFlavorsMatch.call(that, that.entries[entryId], flavorParamsIds);
      return Q.resolve(that.entries[entryId])
}